Helper/DbConnectionParamsNormalizer.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of TheCadien/SuluImportExportBundle.
+ *
+ * (c) Oliver Kossin
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace TheCadien\Bundle\SuluImportExportBundle\Helper;
+
+class DbConnectionParamsNormalizer
+{
+    /**
+     * Normalizes the array with the connection params to a defined structure.
+     *
+     * @param array $connectionParams
+     *        The database connection params for the connection. Should be passed
+     *        as an associative array with the key 'url' or with the keys 'host',
+     *        'user', 'dbname' and 'password'.
+     *        If an URL is used, it should follow the format
+     *        'mysql://<user>:<password>@<host>/<dbname>'.
+     *
+     * @throws \InvalidArgumentException if the passed array does not contain the required fields
+     *
+     * @return array Returns an array with the keys 'host', 'user', 'dbname' and 'password'.
+     *         The latter may be null.
+     */
+    public static function normalize(array $connectionParams): array
+    {
+        if (empty($connectionParams['url'])) {
+            $diff = array_diff_key(['user', 'dbname'], $connectionParams);
+
+            if (\count($diff) > 0) {
+                throw new \InvalidArgumentException(sprintf('The following keys are missing from the $connectionParams: ', implode(', ', $diff)));
+            }
+
+            return [
+                'host' => $connectionParams['host'] ?? 'localhost',
+                'user' => $connectionParams['user'],
+                'dbname' => $connectionParams['dbname'],
+                'password' => $connectionParams['password'] ?? null,
+            ];
```

Comment:
Could be a follow up, but having an object for this, which is then passed to the services would be nice. That way it might be easier to find out where the config in the service is coming from, especially if we want to expand this later, e.g. when we want to check charset from the DoctrineBundle configuration as well.